Controller/CRUDController.php

```diff
@@ -511,7 +511,16 @@ protected function redirectTo($object, Request $request = null)
         }
 
         if (!$url) {
-            $url = $this->admin->generateObjectUrl('edit', $object);
+            foreach (array('edit', 'show') as $route) {
+                if ($this->admin->hasRoute($route) && $this->admin->isGranted(strtoupper($route))) {
+                    $url = $this->admin->generateObjectUrl($route, $object);
+                    break;
```

Comment:
Should it always return the a response? 
This would apply several lines in this method.